server/api/permissions.js

```diff
@@ -106,6 +107,138 @@ const register = function (server, serverOptions) {
       return await Role.create(request.payload);
     }
   });
+
+  server.route({
+    method: 'PUT',
+    path:'/api/permissions/role/{id}',
+    config: {
+      auth: false,
+      pre: [{
+        assign: 'roleValidation',
+        method: function (request,h) {
+
+          const { error } = Joi.validate(request.payload, Role.payload);
+
+          if (error) {
+            throw Boom.badRequest(error.message);
+          }
+
+          return h.continue;
+        }
+      }, {
+        assign: 'permissions',
+        method: async function (request,h) {
+
+          const result = await server.inject({
+            method: 'GET',
+            url: '/api/permissions/available'
+          });
+
+          if (result.statusCode !== 200) {
+            throw Boom.badData(result.message);
+          }
+
+          return result.result;
+        }
+      }, {
+        assign: 'schema',
+        method: function (request,h) {
+
+          return Joi.object().keys(request.pre.permissions.reduce((a, v) => {
+
+            a[v.key] = Joi.boolean();
+            return a;
+          }, {}));
+        }
+      }, {
+        assign: 'validate',
+        method: function (request,h) {
+
+          const { error } = Joi.validate(request.payload.permissions, request.pre.schema);
+
+          if (error) {
+            throw Boom.badRequest(error.message);
+          }
+
+          return h.continue;
+        }
+      }]
+    },
+    handler: async function (request,h) {
+
+      const objectid = request.params.id;
+      const permissions = request.payload.permissions;
+      const name = request.payload.name;
+
+      return await Role.updateRole(objectid,permissions,name);
```

Comment:
Don't need to write a function within the roles model. Just call `await Role.findbyIdandUpdate(id, {$set {...}})`